src/main/java/org/opensearch/security/securityconf/DynamicConfigFactory.java

```diff
@@ -73,8 +73,9 @@
 import org.opensearch.security.support.WildcardMatcher;
 import org.opensearch.threadpool.ThreadPool;
 
-public class DynamicConfigFactory implements Initializable, ConfigurationChangeListener {
+public class DynamicConfigFactory implements ConfigurationChangeListener {
 
+    private final CountDownLatch initializationLatch = new CountDownLatch(1);
```

Comment:
I don't think this latch is working correctly. There are multiple countDowns and the latch is configured to be triggered after a single countDown?

Reply:
agreed, this latch countdown should be removed from onChange. Which is what i intended to get out of discussion on this PR. I have already proposed this as a fix on the original issue: https://github.com/opensearch-project/security/issues/3204#issuecomment-1717993502